src/assets/scripts/client/aircraft/StripView/StripViewModel.js

```diff
@@ -300,48 +356,78 @@ export default class StripViewModel extends BaseModel {
      *
      * @for StripViewModel
      * @method _render
+     * @chainable
      */
     _redraw() {
         return this;
     }
 
-
     /**
+     * Disable the instance ane tear down handlers
      *
-     *
-     * @method reset
+     * @for StripViewModel
+     * @method disable
      */
-    reset() {
+    disable() {
         this.$element.off('click', this.onClickHandler);
         this.$element.off('dblclick', this.onDoubleClickHandler);
+
+        return this;
     }
 
     /**
-     *
+     * Destroy the instance
      *
      * @for StripViewModel
-     * @method shouldUpdate
-     * @param  aircraftModel {AircraftInstanceModel}
-     * @return {boolean}
+     * @method destroy
+     * @chainable
      */
-    shouldUpdate(aircraftModel) {
-        return aircraftModel.fms.currentPhase !== this._flightPhase ||
-            aircraftModel.mcp.altitude !== this._altitude ||
-            aircraftModel.mcp.speed !== this._speed;
+    destroy() {
+        this.disable();
+        this.$element.remove();
+
+        this.id = '';
+        this._eventBus = null;
+        this.$element = null;
+        this.aircraftId = '';
+        this._callsign = '';
+        this._aircraftType = '';
+        this._transponder = 1200;
+        this._assignedAltitude = -1;
+        this._flightPlanAltitude = -1;
+        this._arrivalAirport = '';
+        this._departureAirport = '';
+        this._alternateAirport = '';
+        this._flightPlan = '';
+        this.$callsignView = null;
+        this.$aircraftTypeView = null;
+        this.$transponderView = null;
+        this.$flightPlanAltitudeView = null;
+        this.$arrivalAirportView = null;
+        this.$departureAirportView = null;
+        this.$alternateAirportView = null;
+        this.$flightPlanView = null;
```

Comment:
they are logically ordered;  from the top down, left to right of the `StripView` itself.  this is the same order you will find in `_layout()` and any other method that is dealing with these as a group